.github/workflows/checkstyle.yml

```
@@ -0,0 +1,54 @@
+        uses: actions/cache@v4
+        with:
+          path: ~/.m2/repository
+          key: ${{ runner.os }}-maven-${{ hashFiles('/pom.xml') }}
+          restore-keys: ${{ runner.os }}-maven
+          
+      - name: Install Dependencies
+        run: mvn install -DskipTests
+        
+      - name: Run Checkstyle on Changed Files
+        if: steps.changed-files.outputs.any_changed == 'true'
+        run: |
```

actionlint (1.7.4)

object, array, and null values should not be evaluated in template with ${{ }} but evaluating the value of type {string => string}

(expression)